composition/api/v1alpha1/composition_types.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConditionType defines the type of ManagedConfigSync condition
+type ConditionType string
+
+// The valid conditions of Compositions
+const (
+	Ready ConditionType = "Ready"
+	// Error implies the last reconcile attempt failed
+	Error ConditionType = "Error"
+	// Validation implies the validation failed
+	ValidationFailed ConditionType = "ValidationFailed"
+	// Waiting - Plan is waiting for values to progress
+	Waiting ConditionType = "Waiting"
+)
+
+type Jinja2 struct {
+	Template string `json:"template"`
+}
+
+// ConfigReference - For BYO Expanders, we can extend it
+type ConfigReference struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type ExpanderConfig struct {
+	// Built in expanders
+	Jinja2 *Jinja2 `json:"jinja2,omitempty"`
+	// For BYO Expanders use generic template or ref for external config
+	Template  string           `json:"template,omitempty"`
+	Reference *ConfigReference `json:"configref,omitempty"`
```

Comment:
IMO tend to prefer the struct field name matches the json property name.
```suggestion
	ConfigRef *ConfigReference `json:"configRef,omitempty"`
```

Reply:
Renamed to `ConfigRef`